content/en/blog/posts/2020-11-25-intro-shipwright-pt2.md

```diff
@@ -0,0 +1,190 @@
+---
+title: "Introducing Shipwright - Part 2"
+linkTitle: "Intro to Shipwright (Part 2)"
+description: "A framework for building container images on Kubernetes"
+author: "Adam Kaplan ([@adambkaplan](https://github.com/adambkaplan))"
+date: 2020-11-25T11:53:59-05:00
+draft: false
+---
+
+What is Shipwright? Which problems does this project try to solve?
+
+In [Part 1](/blog/2020/10/21/introducing-shipwright-part-1/) of this series, we looked back at the history of delivering software applications,
+and how that has changed in the age of Kubernetes and cloud-native development.
+
+In this post, we'll introduce Shipwright and the Build APIs that make it simple to
+build container images on Kubernetes.
+
+## Recap - From Binaries to Images
+
+Recall in [Part 1](/blog/2020/10/21/introducing-shipwright-part-1/), developers who moved from 
+VM-based deployments to [Kubernetes](https://kubernetes.io/) needed to shift their unit of delivery
+from binaries to container images. Teams adopted various tools to accomplish this task, ranging
+from [Docker/Moby](https://mobyproject.org/) to [Cloud-Native Buildpacks](https://buildpacks.io/).
+New cloud-native applications like [Jenkins-X](https://jenkins-x.io/) and
+[Tekton](https://tekton.dev/) have emerged to deliver these container images on Kubernetes.
+[Jenkins](https://www.jenkins.io/) - the predecessor to Jenkins-X - has continued to play a 
+prominent role in automating the building and testing of applications.
+
+Building a container image on Kubernetes is not a simple feat. There are a wide variety of tools
+available, some which require knowledge on how to write a Dockerfile, others which do not or are
+tailored for specific programming languages. For CI/CD platforms like Tekton, setting up a build 
+pipeline requires extensive configuration and customization. These are burdens that are difficult
+to overcome for many development teams.
+
+## Shipwright Builds - A Framework for Building Images
+
+With Shipwright, we want to create an open and flexible framework that allows teams easily build
+container images on Kubernetes. Much of the work has been inspired by OpenShift/okd, which provides 
+its own API for building images. At the heart of every build are the following core components:
+
+1. Source code - the "what" you are trying to build
+2. Output image - "where" you are trying to deliver your application
+3. Build strategy - "how" your application is assembled
+4. Invocation - "when" you want to build your application
+
+Developers who use docker are familiar with this process:
+
+1. Clone source from a git-based repository ("what")
+2. Build the container image ("when" and "how")\
```

Comment:
an unnecessary `\` at the end of this sentence.

Reply:
@zhangtbj fixed in the latest push